sirepo/package_data/static/js/cloudmc.js

```diff
@@ -1715,8 +1718,26 @@ SIREPO.app.directive('multiLevelEditor', function(appState, panelState) {
                 }
             }
 
-            function type() {
-                return $scope.model[$scope.field]._type;
+            function type(index) {
+                const f = index == null ? $scope.field : `filter${index}`;
+                return $scope.model[f]._type;
+            }
+
+            function updateEditor() {
+                if ($scope.modelName !== 'filter') {
+                    return;
+                }
```

Comment:
Would it be possible to move the tally.filter specific code out of multiLevelEditor into a view for the tally model? That would keep multiLevelEditor as a general directive, and the filter parts could be contained separately.

Something like this:
```javascript
SIREPO.viewLogic('tallyView', function(panelState, cloudmcService, $scope) {

    function showAvailableFilters() {
        // update filter enum values
    }

    $scope.whenSelected = showAvailableFilters;
    $scope.watchFields = [
        new Array(SIREPO.APP_SCHEMA.constants.maxFilters).fill('').map((_, i) => `tally.filter${i + 1}`),
        showAvailableFilters,
    ];
});
```

Reply:
I'll do that